airflow-core/src/airflow/migrations/versions/0042_3_0_0_add_uuid_primary_key_to_task_instance_.py

```diff
@@ -203,12 +206,28 @@ def upgrade():
     if dialect_name == "postgresql":
         op.execute(pg_uuid7_fn)
 
-        # TODO: Add batching to handle updates in smaller chunks for large tables to avoid locking
         # Migrate existing rows with UUID v7 using a timestamp-based generation
-        op.execute(
-            "UPDATE task_instance SET id = uuid_generate_v7(coalesce(queued_dttm, start_date, clock_timestamp()))"
-        )
-
+        while True:
+            result = conn.execute(
+                text(
+                    f"""
+                    WITH cte AS (
+                        SELECT ctid
+                        FROM task_instance
+                        WHERE id IS NULL
+                        LIMIT {batch_size}
+                    )
+                    UPDATE task_instance
+                    SET id = uuid_generate_v7(coalesce(queued_dttm, start_date, clock_timestamp()))
```

Comment:
I suspect that all the generated uuids will be the same. Can you verify?

Reply:
I ran `select * from task_instance group by id having count(id) > 1` and it returned empty rows

Reply:
![image](https://github.com/user-attachments/assets/5f4b3f4f-c919-401f-8fe8-8b1d75899db8)

Reply:
Oh, yeah cause it makes use of the queued_dttm, start_date of the TI. Thanks